akka-http-core/src/main/scala/akka/http/impl/settings/ConnectionPoolSettingsImpl.scala

```diff
@@ -44,11 +44,21 @@ private[akka] final case class ConnectionPoolSettingsImpl(
   require(minConnections >= 0, "min-connections must be >= 0")
   require(minConnections <= maxConnections, "min-connections must be <= max-connections")
   require(maxRetries >= 0, "max-retries must be >= 0")
-  require(maxOpenRequests > 0 && (maxOpenRequests & (maxOpenRequests - 1)) == 0, "max-open-requests must be a power of 2 > 0")
+  require(maxOpenRequests > 0, "max-open-requests must be a power of 2 > 0.")
+  require((maxOpenRequests & (maxOpenRequests - 1)) == 0, "max-open-requests must be a power of 2. " + suggestPowerOfTwo(maxOpenRequests))
   require(pipeliningLimit > 0, "pipelining-limit must be > 0")
   require(idleTimeout >= Duration.Zero, "idle-timeout must be >= 0")
 
   override def productPrefix = "ConnectionPoolSettings"
+
+  private def suggestPowerOfTwo(around: Int): String = {
```

Comment:
Shouldn't we use a similarly advanced way of figuring out like the above bit-fiddling, just for consistence? ;)

How about something like that [SO answer](https://stackoverflow.com/a/10143264/7647) using [Integer.numberOfLeadingZeros](https://docs.oracle.com/javase/8/docs/api/java/lang/Integer.html#numberOfLeadingZeros-int-).

Reply:
;)

Reply:
it doesn't give correct answers for `around` values that are already powers of two..

Reply:
Heh ;). The point is a bit moot since it is never called with `around` values that are already powers of two.. or do you have an elegant proposal for a replacement?

Reply:
they all involve an extra `if` I'm afraid. :)